src/main/kotlin/net/weavemc/loader/api/event/EventTypes.kt

```diff
@@ -13,28 +13,72 @@ import net.minecraft.world.World
 import org.lwjgl.input.Keyboard
 import org.lwjgl.input.Mouse
 
+/**
+ * This is the base class for all events provided by the Weave Loader.
+ */
 public abstract class Event
+
+/**
+ * This is the base class for all *cancellable* events provided by the
+ * Weave Loader, extending [Event].
+ */
 public abstract class CancellableEvent : Event() {
+
+    /**
+     * This field defines whether the event is cancelled or not. Any mod can cancel and
+     * un-cancel an event. What an event does when cancelled is event-specific, and noted in
+     * that event's documentation.
+     */
     @get:JvmName("isCancelled")
     public var cancelled: Boolean = false
 
 }
 
+/**
+ * This is the base class for a Tick Event. Tick Events can be Pre and Post, but you will never
+ * receive a base Tick Event. These are each called every game tick, or roughly every `50/timer` milliseconds.
```

Comment:
iirc, ticks are typically running at 1/20th of a second, which might be easier to understand in documentation, if you would like to include that.